src/Umbraco.Core/Services/ContentTypeServiceBase.cs

```diff
@@ -71,4 +73,887 @@ internal IEnumerable<IContentTypeBase> GetContentTypesForXmlUpdates(params ICont
 
         }
     }
+
+    internal abstract class ContentTypeServiceBase<TItem, TService> : ContentTypeServiceBase
+        where TItem : class, IContentTypeComposition
+        where TService : class, IContentTypeServiceBase<TItem>
+    {
+        protected ContentTypeServiceBase(IDatabaseUnitOfWorkProvider provider, ILogger logger, IEventMessagesFactory eventMessagesFactory)
+            : base(provider, logger, eventMessagesFactory)
+        {
+            _this = this as TService;
+            if (_this == null) throw new Exception("Oops.");
+        }
+
+        private readonly TService _this;
+
+        public static event TypedEventHandler<TService, SaveEventArgs<TItem>> Saving;
```

Comment:
Not 100% sure either. Yes you can subscribe in two ways but... why would someone do it? It works the way it is... and the real, clean way, would be to get rid of static events which is not a huge task but... later!